infoschema/tables_test.go

```diff
@@ -1218,19 +1201,34 @@ func (s *testTableSuite) TestStmtSummaryHistoryTable(c *C) {
 	tk.MustExec("insert into test_summary    values(2, 'b')")
 	tk.MustExec("insert into TEST_SUMMARY VALUES(3, 'c')")
 	tk.MustExec("/**/insert into test_summary values(4, 'd')")
-	tk.MustQuery(`select stmt_type, schema_name, table_names, index_names, exec_count, sum_cop_task_num, avg_total_keys,
-		max_total_keys, avg_processed_keys, max_processed_keys, avg_write_keys, max_write_keys, avg_prewrite_regions,
-		max_prewrite_regions, avg_affected_rows, query_sample_text
-		from information_schema.statements_summary_history
-		where digest_text like 'insert into test_summary%'`,
-	).Check(testkit.Rows("Insert test test.test_summary <nil> 4 0 0 0 0 0 2 2 1 1 1 insert into test_summary values(1, 'a')"))
+
+	sql := "select stmt_type, schema_name, table_names, index_names, exec_count, sum_cop_task_num, avg_total_keys," +
+		"max_total_keys, avg_processed_keys, max_processed_keys, avg_write_keys, max_write_keys, avg_prewrite_regions," +
+		"max_prewrite_regions, avg_affected_rows, query_sample_text " +
+		"from information_schema.statements_summary_history " +
+		"where digest_text like 'insert into `test_summary`%'"
+	tk.MustQuery(sql).Check(testkit.Rows("Insert test test.test_summary <nil> 4 0 0 0 0 0 2 2 1 1 1 insert into test_summary values(1, 'a')"))
 
 	tk.MustExec("set global tidb_stmt_summary_history_size = 0")
 	tk.MustQuery(`select stmt_type, schema_name, table_names, index_names, exec_count, sum_cop_task_num, avg_total_keys,
 		max_total_keys, avg_processed_keys, max_processed_keys, avg_write_keys, max_write_keys, avg_prewrite_regions,
 		max_prewrite_regions, avg_affected_rows, query_sample_text, plan
 		from information_schema.statements_summary_history`,
 	).Check(testkit.Rows())
+
+	tk.MustExec("set global tidb_enable_stmt_summary = 0")
+	tk.MustExec("drop table if exists `table`")
```

Comment:
I am wondering,  if the original SQL is not quoted but the query result, will that express the intention of test more clearer?

Reply:
`table` is a reserved word, must be quoted.